validations/src/validations.rs

```diff
@@ -367,10 +368,50 @@ pub fn validate_tally_transaction<'a>(
     let tally_stage = &dr_output.data_request.consensus;
 
     validate_consensus(&reveals, &miner_tally, tally_stage)?;
+    validate_tally_outputs(&dr_state, &ta_tx)?;
 
     Ok((ta_tx.outputs.iter().collect(), dr_output.tally_fee))
 }
 
+pub fn validate_tally_outputs(
+    dr_state: &DataRequestState,
+    ta_tx: &TallyTransaction,
+) -> Result<(), failure::Error> {
+    // Expected honest witnesses is tally outputs - 1, which would be
+    // the value transfer output related to the tally change.
+    // In case of tally change doesn't exist and all the outputs are
+    // honest witnesses rewarded, the variable is skipped by the condition
+    // that check if the pkh is equal to de data request pkh.
+    let honest_witnesses = (ta_tx.outputs.len() - 1) as u16;
```

Comment:
Since this variable is only used if `output.pkh == dr_state.data_request.pkh`, declare it inside that if